### DISCUSSION TITLE
Does addon/Properties support sections?

### DISCUSSION THREAD

**Discussion by probonopd** [rsta2/circle#409] [Q&A] [answered]
Does addon/Properties support reading and writing sections? If not, could it be added?

Like this:

```
; Tone Generator 1
[TG1]
Key=Value

; Tone Generator 2
[TG2]
Key=Value
```

Use case: https://github.com/probonopd/MiniDexed/issues/558

**Comment by rsta2** [comment]
Unfortunately it's not supported yet. I'm working hardly for supporting the RPi 5, but when I find some spare time, I will add it. Give me three weeks.

**Comment by rsta2** [comment, marked as answer]
I've updated *addon/Properties* on the *develop* branch to support sections as requested. Please read the commit message for information on how it works. There is a new method `SelectSection()`, which does the job. The support is compatible to the previous solution, when this method is not called. BTW. comments are preceded with `#`. I cannot change this, because `;` may be used as a valid character in values by other users already and this wouldn't be compatible any more then.

**Reply by probonopd** [reply]
Thank you very much @rsta2.
Good to know about `#` for the comments. I was only asking because `;` [is the official comment character](https://en.wikipedia.org/wiki/INI_file#Comments) for `.ini` files, but the same article states that implementations vary and sometimes `#` is used. So I think it is fine.

**Comment by rsta2** [comment]
You are welcome. I wasn't aware of this page about INI files. I don't think, 
that addon/Properties fully meets this specification, but it should be useful 
despite of it.
